### PR TITLE
ADM-794:[backend]fix: add test for pipeline

### PR DESCRIPTION
**Pull request by yulongcai**
## Summary

add test content
## Before

The test coverage is incomplete

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

add test content

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java [hunk 1]
```diff
@@ -104,4 +101,25 @@ void ShouldInitializeMetricsDataCompletedInHandlerWithPreOneWhenPreOneExisted()
...
+	@Test
+	void ShouldInitializeMetricsDataCompletedInHandlerWhenRequestMetricsExist() throws InterruptedException {
+		GenerateReportRequest request = GenerateReportRequest.builder()
+			.csvTimeStamp("csvTimeStamp")
+			.metrics(List.of("velocity", "lead time for changes"))
+			.build();
+		when(asyncMetricsDataHandler.getMetricsDataCompleted(any()))
+			.thenReturn(MetricsDataCompleted.builder().pipelineMetricsCompleted(true).build());
+		doAnswer(invocation -> null).when(generateReporterService).generateBoardReport(request);
+		reportService.generateReportByType(request, MetricType.BOARD);
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/ReportServiceTest.java:113]: new line for `given when then`

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java [hunk 2]
```diff
@@ -104,4 +101,25 @@ void ShouldInitializeMetricsDataCompletedInHandlerWithPreOneWhenPreOneExisted()
...
+		Thread.sleep(100);
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/ReportServiceTest.java:114]: try to remove sleep to improve the speed of UT
Maybe you can mock more things

File: backend/src/test/java/heartbeat/service/report/PipelineServiceTest.java [hunk 3]
```diff
@@ -61,20 +62,23 @@ class FetchGithubData {
 
 		@Test
 		void shouldReturnEmptyBuildInfosListAndEmptyLeadTimeWhenDeploymentEnvironmentsIsEmpty() {
+			String token = "token";
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/PipelineServiceTest.java:65]: using fixture, not customize

File: backend/src/test/java/heartbeat/service/report/KanbanServiceTest.java [hunk 4]
```diff
@@ -33,8 +33,11 @@ class KanbanServiceTest {
 
 	@Test
 	void shouldCallCsvServiceToGenerateScvInfo() {
+		JiraBoardSetting mockJiraBoardSetting = KanbanFixture.MOCK_JIRA_BOARD_SETTING();
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/KanbanServiceTest.java:36]: for the test description `shouldCallCsvServiceToGenerateScvInfo` -> `shouldCallCsvServiceToGenerateCSVInfo`, yes?

Comment (guzhongren) [reply backend/src/test/java/heartbeat/service/report/KanbanServiceTest.java:36]: what is `when` for this test case

File: backend/src/test/java/heartbeat/service/report/KanbanFixture.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+public class KanbanFixture {
+
+	public static JiraBoardSetting MOCK_JIRA_BOARD_SETTING() {
+		return JiraBoardSetting.builder()
+			.users(List.of("user1"))
+			.token("token")
+			.type("jira")
+			.site("site")
+			.projectKey("ADM")
+			.boardId("2")
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/KanbanFixture.java:19]: duplicated code with line 29 ~ line 33

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java [hunk 6]
```diff
@@ -104,4 +101,25 @@ void ShouldInitializeMetricsDataCompletedInHandlerWithPreOneWhenPreOneExisted()
 		verify(generateReporterService, never()).generateDoraReport(request);
 	}
 
+	@Test
+	void ShouldInitializeMetricsDataCompletedInHandlerWhenRequestMetricsExist() throws InterruptedException {
+		GenerateReportRequest request = GenerateReportRequest.builder()
+			.csvTimeStamp("csvTimeStamp")
+			.metrics(List.of("velocity", "lead time for changes"))
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/report/ReportServiceTest.java:108]: do we have enum for `("velocity", "lead time for changes"`

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java (hunk 2 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/ReportServiceTest.java]: done

File: backend/src/test/java/heartbeat/service/report/PipelineServiceTest.java (hunk 3 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/PipelineServiceTest.java]: done

File: backend/src/test/java/heartbeat/service/report/KanbanServiceTest.java (hunk 4 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/KanbanServiceTest.java:36]: done

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java (hunk 6 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/ReportServiceTest.java]: done

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java (hunk 1 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/ReportServiceTest.java:116]: when it's too long to pass e2e.

File: backend/src/test/java/heartbeat/service/report/KanbanFixture.java (hunk 5 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/KanbanFixture.java:19]: different types of object definitions, do not know how to repeat

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java (hunk 1 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/ReportServiceTest.java:116]: Sorry, I didn't get it. Change it right away

**Review by mikeyangyun — APPROVED** [review]

File: backend/src/test/java/heartbeat/service/report/ReportServiceTest.java (hunk 1 above)
Comment (yulongcai) [reply backend/src/test/java/heartbeat/service/report/ReportServiceTest.java:127]: done

**Pull request merged**